substra/sdk/backends/local/backend.py

```diff
@@ -399,7 +399,11 @@ def _add_traintuple(self, spec, exist_ok, spec_options=None):
         )
 
         traintuple = self._db.add(traintuple, exist_ok)
-        self._worker.schedule_traintuple(traintuple)
+        fake_data = spec_options is not None and spec_options.get("fake_data", False)
+        n_fake_samples = spec_options.get("n_fake_samples")
```

Comment:
Shouldn't you test that spec options is defined here as well?

Reply:
Also, I think you're missing fake data sample handling for composite traintuples.